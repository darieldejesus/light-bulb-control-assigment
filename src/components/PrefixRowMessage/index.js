import React from 'react';
import PropTypes from 'prop-types';
import { Txt } from 'rendition';

/**
 * Component to show a row with the given message in a table.
 *
 * @param {Object} obj - Main parameter.
 * @param {number} obj.colSpan - Column span.
 * @param {string} obj.message - Message to be shown.
 * @returns {Object} ReactElement.
 * */
export default function PrefixRowMessage({ colSpan, message }) {
  return (
    <tr>
      <td colSpan={colSpan}>
        <Txt.p align="center">
          {message}
        </Txt.p>
      </td>
    </tr>
  );
}

PrefixRowMessage.propTypes = {
  colSpan: PropTypes.number.isRequired,
  message: PropTypes.string.isRequired,
};
