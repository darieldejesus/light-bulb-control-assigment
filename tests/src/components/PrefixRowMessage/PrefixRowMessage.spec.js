import React from 'react';

import PrefixRowMessage from '../../../../src/components/PrefixRowMessage';

describe('PrefixRowMessage', () => {
  test('Should render a table row with a given message', () => {
    const wrapper = shallow(<PrefixRowMessage colSpan={1} message="Hello World!" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
