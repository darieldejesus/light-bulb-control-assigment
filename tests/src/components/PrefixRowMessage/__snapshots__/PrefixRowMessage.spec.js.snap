// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrefixRowMessage Should render a table row with a given message 1`] = `
<tr>
  <td
    colSpan={1}
  >
    <Component
      align="center"
    >
      Hello World!
    </Component>
  </td>
</tr>
`;
